chainstate/src/detail/transaction_verifier/mod.rs

```diff
@@ -489,6 +531,29 @@ impl<'a, S: BlockchainStorageRead> TransactionVerifier<'a, S> {
         Ok(())
     }
 
+    fn register_tokens(&mut self, tx: &Transaction) -> Result<(), TokensError> {
+        let token_id = token_id(tx).ok_or(TokensError::TokenIdCantBeCalculated)?;
+
+        let _tokens_op = match self.tokens_cache.entry(token_id) {
+            Entry::Occupied(entry) => entry.into_mut(), // TODO: Tokens already issued? Must we return error?
+            Entry::Vacant(entry) => entry.insert(CachedTokensOperation::Write(tx.get_id())),
+        };
+        Ok(())
+    }
+
+    fn undo_issuance(&mut self, token_id: TokenId) -> Result<(), TokensError> {
+        match self.tokens_cache.entry(token_id) {
+            Entry::Occupied(entry) => {
+                let tokens_op = entry.into_mut();
+                *tokens_op = CachedTokensOperation::Erase;
+            }
+            Entry::Vacant(entry) => {
+                entry.insert(CachedTokensOperation::Erase);
```

Comment:
Again, this is a consistency issue. It's IMPOSSIBLE to disconnect a block with token issuance without finding a token issued. Please don't hide such issues and handle them properly.

Reply:
Fixed. If we can't find tokens here, we return `TokensError::InvariantBrokenUndoIssuanceOnNonexistentToken`